File: backend/app/api/v1/endpoints/predictions.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from ....db.session import get_db
from ....schemas.prediction import HistoryResponse, PredictionPoint
from ....services import prediction_service

router = APIRouter(prefix="/predictions", tags=["predictions"])


@router.get("/{disease}/{iso3}", response_model=PredictionPoint)
def get_prediction(
    disease: str,
    iso3: str,
    year: int = Query(..., ge=2022, le=2030),
    week: int = Query(..., ge=1, le=53),
    db: Session = Depends(get_db),
):
    return prediction_service.get_prediction(db, disease, iso3, year, week)


@router.get("/{disease}/{iso3}/history", response_model=HistoryResponse)
def get_history(
    disease: str,
    iso3: str,
    start_year: int = Query(2022, ge=2010),
    end_year: int = Query(2022, le=2030),
    db: Session = Depends(get_db),
):
    return prediction_service.get_history(db, disease, iso3, start_year, end_year)

File: frontend/src/api/predictions.ts
import api from "./axios";
import type { HistoryResponse, PredictionPoint } from "../types/api";

export async function fetchPrediction(
  disease: string,
  iso3: string,
  year: number,
  week: number,
): Promise<PredictionPoint> {
  const { data } = await api.get<PredictionPoint>(`/predictions/${disease}/${iso3}`, {
    params: { year, week },
  });
  return data;
}

export async function fetchHistory(
  disease: string,
  iso3: string,
  startYear: number,
  endYear: number,
): Promise<HistoryResponse> {
  const { data } = await api.get<HistoryResponse>(`/predictions/${disease}/${iso3}/history`, {
    params: { start_year: startYear, end_year: endYear },
  });
  return data;
}
